src/session/session_dhandle.c

```diff
@@ -229,7 +229,8 @@ __wt_session_lock_dhandle(
 			WT_ASSERT(session, !F_ISSET(dhandle, WT_DHANDLE_DEAD));
 			return (0);
 		}
-		if (ret != EBUSY || (is_open && want_exclusive))
+		if (ret != EBUSY || (is_open && want_exclusive) ||
+		    LF_ISSET(WT_DHANDLE_LOCK_ONLY))
```

Comment:
Looks to me as though ``WT_DHANDLE_LOCK_ONLY`` is set when calling ``__wt_session_lock_dhandle`` for checkpoints as well - is changing the semantic for checkpoints an expected consequence.

Answering my own question.. checkpoint is asking for exclusive access, so would already take this path.